### PR TITLE
Send `data-update` event when text changes

### PR DESCRIPTION
**Pull request by olegrise**
## Description
Send `data-update` event when text changes. Here is an example of event consumption:
```
  element.addEventListener('data-update', function (event) {
    console.log(event.type); // prints 'data-update'
    console.log(event.detail.newValue); // prints new text value
    console.log(event.detail.oldValue); // prints old (previous) text value
  });
```

## Motivation and Context
https://trello.com/c/qGDkIhPX/207-1-rise-text-add-change-listener

## How Has This Been Tested?
Automated testing. Unit tests were updated to cover new functionality. 

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Comment by olegrise** [comment]
@andrecardoso please review. Couple of comments:

- Do you think we should increase the version to 2.0.0?
- I'm going to ask Peter to validate this change before deploying to stable.

**Comment by olegrise** [comment]
@pcsandford I just realized that I should probably ask you - do you prefer that we keep the rise-text version the same or bump it up for this release?

File: README.md [hunk 1]
```diff
@@ -41,6 +41,14 @@ This component receives the following list of attributes:
 The component sends the following events:
 
 - **_configured_**: The component has initialized what it requires to and is ready to handle a _start_ event.
+- **_data-update_**: Event is sent when text value changes. Here is an example of event consumption:
+```
+  element.addEventListener('data-update', function (event) {
+    conslove.log(event.type); // prints 'data-update'
```
Comment (andrecardoso) [inline README.md]: `conslove` typo ❤️

Comment (olegrise) [reply README.md]: @andrecardoso fixed :)

**Pull request merged**